src/ticTacToeGame.java

```diff
@@ -16,15 +17,72 @@ public class ticTacToeGame {
 
     public static void main(String[] args) {
         // TODO Auto-generated method stub
-
-        symbol_choosing();
+        boolean gameStarted = checkIfGameExists();
+        if (!gameStarted) {
+            loadGame();
+        } else {
+            symbol_choosing();
+        }
+        
         while (!game_ended()) {
             print_board();
             place = ask_move(player);
             add_board_move(player, place);
             change_player();
+            saveGameState();
+        }
+
+    }
+
+    private static void loadGame() {
+        // TODO Auto-generated method stub
+        File gameState = new File("data/gameState.csv");
+        
+        try {
+            Scanner stateScanner = new Scanner(gameState);
+            player = Integer.parseInt(stateScanner.nextLine());
+            player1Symbol = stateScanner.nextLine().charAt(0);
+            player2Symbol = stateScanner.nextLine().charAt(0);
+            
+            for(int i = 0; i < board.length; ++i) {
+                for(int j = 0; j < board[0].length; ++j) {
+                    board[i][j] = stateScanner.nextLine().charAt(0);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+    }
+
+    private static boolean checkIfGameExists() {
+        // TODO Auto-generated method stub
+        File gameState = new File("data/gameState.csv");
+        
+        if (gameState.exists()) {
+            return true;
+        } else {
+            return false;
         }
+        
+    }
 
+    public static void saveGameState() {
+        // TODO Auto-generated method stub
+        try {
+            FileWriter gameState = new FileWriter("data/gameState.csv");
```

Comment:
use meanfingfull name for file name

Reply:
This is a meaningful name, it means the last game state.